clang/test/CIR/CodeGen/builtin-arm-ldrex.c

```diff
@@ -0,0 +1,21 @@
+// RUN: %clang_cc1 -triple aarch64-none-linux-android24  -fclangir -emit-cir -mmlir --mlir-print-ir-before=cir-lowering-prepare %s -o %t.cir 2>&1 | FileCheck %s
+int foo() { int a; return __builtin_arm_ldrex(&a); }
+void bar() { int a; int* p = (int*)__builtin_arm_ldrex(&a); }
+// CHECK: cir.func no_proto  @foo() -> !s32i
+// CHECK:  [[RET_VAL:%.*]] = cir.alloca !s32i, !cir.ptr<!s32i>, ["__retval"] {alignment = 4 : i64}
+// CHECK:  [[A:%.*]] = cir.alloca !s32i, !cir.ptr<!s32i>, ["a"] 
+// CHECK:  [[INTRN_RET:%.*]] = cir.llvm.intrinsic "llvm.aarch64.ldxr" [[A]] : (!cir.ptr<!s32i>) -> !s64i
+// CHECK:  [[RET_CAST:%.*]] = cir.cast(integral, [[INTRN_RET]] : !s64i), !s32i
+// CHECK:  cir.store [[RET_CAST]], [[RET_VAL]] : !s32i, !cir.ptr<!s32i>
+// CHECK:  [[RET:%.*]] = cir.load [[RET_VAL]] : !cir.ptr<!s32i>, !s32i
+// CHECK:  cir.return [[RET]] : !s32i 
```

Comment:
We cannot yet add LLVM check, but it'd be nice to leave a snippet around so we can double check it's doing the right job when we introduce it, example:
```
// This is how LLVM output should look like:
//  %2 = call i64 @llvm.aarch64.ldxr.p0(ptr elementtype(i32) %1)
//  %3 = trunc i64 %2 to i32
```
Don't need to do this everywhere, only one per covered type.